formver.compiler-plugin/core/src/org/jetbrains/kotlin/formver/core/conversion/ProgramConverter.kt

```diff
@@ -120,9 +125,23 @@ class ProgramConverter(
 
         // Note: it is important that `body` is only set after `embedUserFunction` is complete, as we need to
         // place the embedding in the map before processing the body.
-        embedUserFunction(declaration.symbol, signature).apply {
-            body = stmtCtx.convertMethodWithBody(declaration, signature, returnTarget)
+        if (declaration.symbol.isPure(session)) {
```

Comment:
I'm afraid this can't be the right place to do this.  Namely: this is only called for the "root" function that we are verifying, while you want to add bodies to all pure functions (those bodies are visible in methods that call them).  So you want to hook this up in a place that is called for all functions a user calls.